src/firefly/js/ui/FileUpload.jsx

```diff
@@ -89,7 +99,14 @@ FileUploadView.defaultProps = {
 export const FileUpload= memo( (props) => {
     const {viewProps, fireValueChange}=  useFieldGroupConnector(props);
     let modViewProps;
-    if (viewProps.isFromURL) {
+
+    useEffect(() => {
+        if (viewProps.isFileDrop) {
+            handleChange(viewProps.dropEvent, fireValueChange, viewProps.fileType, viewProps.fileAnalysis, viewProps.setFileDrop);
+        }
```

Comment:
is approach is fine. Just don't use isFileDrop use `dropEvent`. I think you can get away with not using get/setFileDrop at all.